net/golioth/system_client.c

```diff
@@ -283,14 +299,17 @@ static void golioth_system_client_main(void *arg1, void *arg2, void *arg3)
 
 		if (timeout_occurred) {
 			bool reconnect_request = atomic_test_and_clear_bit(&flags, FLAG_RECONNECT);
+			bool stop_request = atomic_test_and_clear_bit(&flags, FLAG_STOP_CLIENT);
 			bool receive_timeout = (recv_expiry <= k_uptime_get());
 
 			/*
-			 * Reconnect request is handled similar to recv timeout.
+			 * Reconnect and stop requests are handled similar to recv timeout.
 			 */
-			if (reconnect_request || receive_timeout) {
+			if (reconnect_request || receive_timeout || stop_request) {
 				if (reconnect_request) {
 					LOG_INF("Reconnect per request");
+				} else if (stop_request) {
+					LOG_INF("Stop request");
```

Comment:
Minor nit. Assuming that both "stop" and "reconnect" will be handled at the same time, then the "stop" will take precedence. So I would reorder if statements here and do:
```suggestion
				if (stop_request) {
					LOG_INF("Stop request");
				} else if (reconnect_request) {
					LOG_INF("Reconnect per request");
```

Reply:
Good point. Change addressed!